### PR TITLE
Update Documentation

### PR DESCRIPTION
**Pull request by zewolfe**
This PR updates the documentation to reflect changes and additions such as the cluster reconciler.

Issue - https://github.com/giantswarm/roadmap/issues/1370

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,32 +3,42 @@ Operator to automate workload identity setup on GCP clusters
...
 > Pods that use the configured Kubernetes service account automatically authenticate as the IAM service account when accessing Google Cloud APIs.
 > Using Workload Identity allows you to assign distinct, fine-grained identities and authorization for each application in your cluster.
 
+## Concepts
+
+This operator is set up to work on a multi-cluster setup. What is a multi-cluster setup?
+
+> Multi-cluster Kubernetes is a Kubernetes deployment method that consists of two or more clusters.
+
+It operates on the notion of management and workload clusters. Where a management cluster manages the lifecycle of a workload cluster and vice versa.
```
Comment (nprokopic) [inline README.md:20]: nitpick :grimacing: 

> Where a management cluster manages the lifecycle of a workload cluster and vice versa.

"and vice versa"? Would that mean that a workload cluster manages the lifecycle of a management cluster?

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,15 +8,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ### Changed
-
+- Updated documentation
 - Increase container memory resource requests and limits
 
 ## [0.4.0] - 2022-09-21
 
-### Added
```
Comment (AverageMarcus) [inline CHANGELOG.md:16]: Did you mean to remove this?

Comment (zewolfe) [reply CHANGELOG.md:16]: :( I did not. Must've happened during conflict resolution.

File: README.md (hunk 1 above)
Comment (zewolfe) [reply README.md]: 😄 No, I mean't that a workload cluster has their lifecycle managed by a management cluster. 
I've updated it to be more akin to that found here - https://cluster-api.sigs.k8s.io/user/concepts.html

File: README.md [hunk 3]
```diff
@@ -86,23 +126,31 @@ gcloud container hub memberships describe $MEMBERSHIP_NAME
     --member="serviceAccount:$WORKLOAD_ID_POOL[$KUBE_NAMESPACE/$KUBE_SA_NAME]"
 ```
 
-##### 3.5 Add computer viewer permissions
+##### 4 Add computer viewer permissions
```
Comment (mnitchev) [inline README.md:129]:
```suggestion
##### 4 Ensure that your GCP service account has the roles that the workload will need.

Example: Add the `compute.viewer` role:
```

File: README.md [hunk 4]
```diff
@@ -57,23 +106,14 @@ kubectl create serviceaccount $KUBE_SA_NAME \
...
-These are the following:
-  * Your workload identity pool id
-  * Your identity provider 
-The above can be obtained from the output of the command below
-```
-gcloud container hub memberships describe $MEMBERSHIP_NAME
-```
-
-##### 3.4 Ensure that your gcp service account has the roles that you need. 
+##### 3 Ensure that your GCP service account has the roles that you need.
```
Comment (mnitchev) [inline README.md:116]:
```suggestion
##### 3 Give the Kubernetes Service Account permission to impersonate the GCP Service Account
```

**Review by mnitchev — APPROVED** [review]

**Pull request merged**